OpenRA.Mods.Cnc/Effects/GpsDotEffect.cs

```diff
@@ -80,6 +80,9 @@ bool ShouldRender(DotState state, Player toPlayer)
 			if (!toPlayer.Shroud.IsExplored(actor.CenterPosition))
 				return false;
 
+			if (actor.Trait<GpsDot>().IsTraitDisabled)
```

Comment:
I think the trait lookup should be cached.

Reply:
Done.